numba/tests/test_np_functions.py

```diff
@@ -296,6 +299,32 @@ def check(x_types, x_values):
         x_types = [types.complex64, types.complex128]
         check(x_types, x_values)
 
+    def test_delete(self):
+
+        def arrays():
+            # array, obj, axis
+            # 
+            # 1d array, scalar, None
+            yield np.arange(10), 3, None
+            yield np.arange(10), -3, None # Negative obj
+            # 1d array, list, None
+            yield np.arange(10), [3, 5, 6], None
+            yield np.arange(10), [2, 3, 4, 5], None
+            # 3d array, scalar, None
+            yield np.arange(3 * 4 * 5).reshape(3, 4, 5), 2, None
+            # 3d array, list, None
+            yield np.arange(3 * 4 * 5).reshape(3, 4, 5), [5, 30, 27, 8], None
+            #
+            # yield np.arange(5 * 2).reshape(5, 2), 2, 0
+
+        pyfunc = delete
+        cfunc = jit(nopython=True)(pyfunc)
+
+        for a, obj, axis in arrays():
+            expected = pyfunc(a, obj, axis)
+            got = cfunc(a, obj, axis)
+            self.assertPreciseEqual(expected, got)
+
```

Comment:
Some tests need adding to check that the correct exceptions are raised on invalid input. This sort of pattern would likely be appropriate (i.e. check exception type and message):

https://github.com/numba/numba/blob/0ede0b7e8dc3ec8f7d22210573fbf268bf0d68f6/numba/tests/test_dictimpl.py#L533-L538

Reply:
I have tried the approach below but the test is failing because of a memory leak.

```python
with self.assertRaises(IndexError) as raises:
            cfunc([1, 2], 3)
self.assertIn(
    'delete(): pos must be less than the len(arr)',
    str(raises.exception)
)
```

```python
======================================================================
FAIL: test_delete (numba.tests.test_np_functions.TestNPFunctions)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/guilhermeleobas/Documents/GitHub/numba/numba/tests/support.py", line 645, in tearDown
    self.memory_leak_teardown()
  File "/Users/guilhermeleobas/Documents/GitHub/numba/numba/tests/support.py", line 619, in memory_leak_teardown
    self.assert_no_memory_leak()
  File "/Users/guilhermeleobas/Documents/GitHub/numba/numba/tests/support.py", line 628, in assert_no_memory_leak
    self.assertEqual(total_alloc, total_free)
AssertionError: 36 != 35
```

Do you have any suggestion on how can I address this issue?

Reply:
Testing for Exceptions will lead to a leak. It is best to test all exceptions in a separate function, e.g. `test_delete_exceptions` and then use the following to suppress the leak check.

`self.disable_leak_check()`

Reply:
Here is an example:

https://github.com/numba/numba/blob/master/numba/tests/test_np_functions.py#L2590

Reply:
Thanks, that solved the issue.